mlx_vlm/models/qwen2_5_vl/qwen2_5_vl.py

```diff
@@ -89,18 +89,107 @@ def get_input_embeddings(
 
     @staticmethod
     def merge_input_ids_with_image_features(
-        image_token_id, video_token_id, image_features, inputs_embeds, input_ids
+        image_token_id,
+        video_token_id,
+        image_features,
+        inputs_embeds,
+        input_ids,
     ):
-        image_token_id = image_token_id
-        video_token_id = video_token_id
-        # Positions of <image> tokens in input_ids, assuming batch size is 1
+        """Merge image features into input embeddings at image token positions.
+
+        Args:
+            image_token_id: The token ID for image placeholders
+            video_token_id: The token ID for video placeholders (fallback)
+            image_features: Vision features from the vision tower [num_features, hidden_dim]
+            inputs_embeds: Input embeddings [batch_size, seq_len, hidden_dim]
+            input_ids: Input token IDs [batch_size, seq_len]
+            grid_thw: Grid dimensions for each image (optional, not used in simple case)
+
+        Returns:
+            Updated input embeddings with image features inserted
+        """
+        # Find positions of image tokens
         image_positions = input_ids == image_token_id
         if mx.sum(image_positions) == 0:
             image_positions = input_ids == video_token_id
 
-        image_indices = np.where(image_positions)[1].tolist()
-        inputs_embeds[:, image_indices, :] = image_features
-        return inputs_embeds
+        # Get all positions where we need to insert image features
+        batch_size, seq_len = input_ids.shape
+
+        # For batch size 1 (common case), we can do direct assignment
+        if batch_size == 1:
```

Comment:
Can we simply this part?

Perhaps keep the batch size of N logic or merge the batch size of 1 and N into one.